engine/target_sym_engine.ml

```diff
@@ -147,15 +154,20 @@ let rec subst_svalue bindings = function
 let rec subst_tree bindings = function
   | Failure -> Failure
   | Leaf result -> Leaf result
-  | Node children ->
+  | Node (children, fallback) ->
      let subst_pi bindings {var; op} =
        {var = subst_svalue bindings var; op}
      in
-     let subst (pis, tree) =
-       (List.map (subst_pi bindings) pis,
+     let subst (pi, tree) =
+       (subst_pi bindings pi,
         subst_tree bindings tree)
      in
-     Node (List.map subst children)
+     let fallback' = match fallback with
+       | None -> None
+       | Some (pis, tree) ->
+         Some ((List.map (subst_pi bindings) pis), subst_tree bindings tree)
+     in
+     Node ((List.map subst children), fallback')
```

Comment:
This could also be written as

```ocaml
let subst_pi (pi, tree) = ... in
let subst_pis (pis, tree) = ... in
Node(List.map subst_pi children, Option.map subst_pis fallback)
```

Reply:
Done in https://github.com/FraMecca/inria-internship/pull/15/commits/555cfe7c4a776755ea51aa3c7b0349b8b8b5c4f8